### PR TITLE
Add configurable print formatting types

### PR DESCRIPTION
**Pull request by tmartin-gh**

### CODE REVIEW

File: examples/print_styles.cu [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  const complex Ainit[16] {
+    {  1,  -1}, {  2,   2}, {  3,  -3}, {  4,  4},
+    { -5,   5}, { -6,  -6}, { -7,   7}, { -8, -8},
+    {  9,  -9}, { 10,  10}, { 11, -11}, { 12, 12},
+    {-13,  13}, {-14,  14}, {-15,  15}, {-16, 16}};
+
+  auto A1 = make_tensor<complex>({16});
+  for (int k=0; k<16; k++)
+  {
+    A1(k) = Ainit[k];
```
Comment (cliffburdick) [inline examples/print_styles.cu:61]: You should be able to replace this loop with `A1.SetVals()`

File: include/matx/core/tensor_utils.h [hunk 2]
```diff
@@ -606,54 +622,165 @@ namespace detail {
...
-        PrintVal(op.operator()(_k));
-        printf("\n");
+        if (_k == 0)
+        {
+          if ((PRINT_FORMAT_TYPE == MATX_PRINT_FORMAT_MLAB) || (PRINT_FORMAT_TYPE == MATX_PRINT_FORMAT_PYTHON))
+          {
+            fprintf(fp, "[");
+          }
+        }
+        if (PRINT_FORMAT_TYPE == MATX_PRINT_FORMAT_DEFAULT) fprintf(fp, "%06" INDEX_T_FMT ": ", _k);
```
Comment (cliffburdick) [inline include/matx/core/tensor_utils.h:646]: The unofficial style in the rest of the code is to have braces around any `if`, even if it's one line. Can you update those places?

File: include/matx/core/tensor_utils.h [hunk 3]
```diff
@@ -606,54 +622,165 @@ namespace detail {
...
+          if ((PRINT_FORMAT_TYPE == MATX_PRINT_FORMAT_MLAB) || (PRINT_FORMAT_TYPE == MATX_PRINT_FORMAT_PYTHON))
+          {
+            fprintf(fp, "]");
+          }
+        }
+        else
+        {
+          if (PRINT_FORMAT_TYPE == MATX_PRINT_FORMAT_PYTHON)
+          {
+            fprintf(fp, ",");
```
Comment (cliffburdick) [inline include/matx/core/tensor_utils.h:659]: I think MATLAB allows commas between elements too. Can this be simplified by always adding it?

**Review by cliffburdick — CHANGES_REQUESTED** [review]
Thanks!

File: examples/print_styles.cu (hunk 1 above)
Comment (tmartin-gh) [reply examples/print_styles.cu]: Done.

File: include/matx/core/tensor_utils.h (hunk 2 above)
Comment (tmartin-gh) [reply include/matx/core/tensor_utils.h:646]: Done.  Moved if/else opening brace to same line where I saw it.

File: include/matx/core/tensor_utils.h (hunk 3 above)
Comment (tmartin-gh) [reply include/matx/core/tensor_utils.h:659]: Changed to always add for python and matlab

**Review by cliffburdick — APPROVED** [review]

**Pull request merged**